RobotsAndPencils/slack-salesforce-bot
# Renamed to SalesforceOpportunityEvents and Message as Attachment

#### What's this PR do?
Refactored PubSubEvent* to SalesforceOpportunityEvent* in order to create a better conceptual mapping, since the PubSub topic/subscription is only related to when an Opportunity goes into the Contracts in Legal state.  Also, beautified the bot's message of the event by creating a Slack attachment with the relevant Opportunity data.

#### Where should the reviewer start?
See changelist.

#### How should this be manually tested?
`source setenv.sh`
`gb build all`
`./bin/bot`

Go into our Salesforce sandbox instance and edit or create a new opportunity and then make sure its in the Contracts in Legal stage.  Save changes and then look for the bot to post a message as an Attachment to any slack channels he's a member of.

#### Screenshots (if appropriate)
![image](https://cloud.githubusercontent.com/assets/1517743/13227856/c5adf7d6-d966-11e5-8955-21451a500e3a.png)
